### ISSUE TITLE
Enable any callable to be a criterion or skip criterion section if loss is returned in `run_step`

### ISSUE THREAD

**Issue by jamesWalczak** [opengeokube/kit4dl#37]
**Is your feature request related to a problem? Please describe.**
Currently only subclasses of `torch.nn.Module` can be criterions. 
If loss value is returned in the `run_step`-like methods, the criterion configuration is redundant (it's up to a user then to compute the loss properly).

**Describe the solution you'd like**
Besides the current logic, also any function could be supported.
If criterion section was not provided in the configuration file, `run_step` or `run_train_step` should return also loss value.
